factorio-map-process.sh

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+set -euo pipefail
+## Defaults
+SERVER_BASE_PATH=/srv/factorio/maps
+MAP_TILES_PATH=images
+FACTORIO_MAP_SCRIPT="$(dirname "${BASH_SOURCE[0]}")/factoriomap.py"
+TARFILE=
+REDUCE=0
+
+function usage() {
+    cat <<EOF
+USAGE
+
+factorio-map-process.sh [OPTIONS] tarfile
+
+DESCRIPTION
+
+This script will process a given tarfile into a set of images used in a leaflet map.
+The default behavior will create a directory for the world maps if one does not exist and copy the example html page in as 'index.html'
+The tar file will be parsed for a map name and a date, separated by underscores (_). Files should be of the format {WorldName}_YYYY-MM-DD.tar.
+    
+OPTIONS
+
+    -t | --tiles <images>           Set the name of the directory to store the tile images in.  Default value is
+                                    'images'
+
+    -s | --server <server path>     Set the base path where the http server files will be stored.  Default value is
+                                    '/srv/factorio/maps'
+
+    -r                              Flag to size-reduce the images folder with the rdfind tool
+EOF
+}
+
+function testRequirements() {
+	if ! [ -x "$(command -v jq)" ]; then
+		echo "ERROR: Requirement not satisfied - jq -"
+		exit 1
+	fi
+
+        if [[ REDUCE -eq 1 && !(-x "$(command -v rdfind)") ]]; then
+		echo "ERROR: Requirement not satisfied - rdfind -"
+		exit 1
+	fi
+
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FILENAME%.*}"
+	echo "Target File: ${FILENAME}"
+
+	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< $NAME_NOEXT))
+
+	WORLDNAME=${infos[0]}
+	echo "WORLD NAME $WORLDNAME"
+
+	DATESTR=${infos[1]}
+	echo "DATE STRING $DATESTR"
```

Comment:
I would still like to be able to support only one world. That is, not having a world directory under my base path.

Maybe something like this could be done here?
```suggestion
	if [[ ${#infos[@]} -ge 2 ]]; then
		WORLDNAME="${infos[0]}"
		echo "WORLD NAME $WORLDNAME"

		DATESTR=${infos[1]}
		echo "DATE STRING $DATESTR"
	else
		WORLDNAME=""
		DATESTR=${infos[0]}
		echo "DATE STRING $DATESTR"
	fi
```
By setting `WORLDNAME` to an empty string, we are setting it, which will prevent errors with the `set -u` at the start of the script. But being empty, when we use it to create a path, that dir will be skipped.

Reply:
I like this solution.